ZNTA-939 Add support for wildfly 10

seanf opened the pull request:
https://zanata.atlassian.net/browse/ZNTA-939

Also add logging filter configuration for some unavoidable startup warnings.

seanf commented:
@zanata-jenkins retest this please

seanf commented:
@zanata-jenkins retest this please

seanf commented:
Tests passed on EAP 6; now rerunning on WildFly.

carlosmunoz commented:
Seems to be all-config. Nothing to object.
:+1:

carlosmunoz commented:
I take that back. Needs documentation on it now running on WF 10

seanf commented:
> Needs documentation on it now running on WF 10

Okay, but can I get you to update zanata-overlay (including standalone.xml) for WF 10?

seanf commented:
@carlosmunoz Happy with commit 7c2c9ac5a9113db6bd94730d0888b56a42dfc24f?

carlosmunoz commented:
:+1: on the docs. I will add a commit for the standalone.xml file in the layover

seanf commented:
@carlosmunoz Layover? Are you flying somewhere?  Oh, overlay. :-)

Thanks.

carlosmunoz commented:
@seanf Added the wildfly 10 standalone ... I'd like QA to have a go at installing this

seanf commented:
@zanata-jenkins test this please

seanf commented:
The build failed, but only because the Jenkins workspace cleanup plugin had a problem cleaning up at the end.  All the tests passed, both on EAP 6 and WildFly 10.

@carlosmunoz Do you think we should go ahead with this now, or wait until we can do something about the script standalone.cli.messaging.config?

carlosmunoz commented:
I'm not sure I recall what the standalone.cli.messaging.config script is about.

I also never did the rename of the wildfly-10 profile to simply wildfly. :fearful: 

But in any case, the earlier we can start testing this the better. So unless there is a powerful reason not to integrate, I say we go for it.

seanf commented:
@carlosmunoz I think standalone.cli.messaging.config is a jboss-cli script which modifies the JMS config in standalone.xml to suit Zanata, but it would need a complete rewrite for WildFly 10. Right @huangp?

seanf commented:
@huangp Could you please review the last commit "Update JMS config docs/script for WildFly 10"?

seanf commented:
@zanata-jenkins test this please

seanf commented:
@carlosmunoz Happy with 373c39c?

seanf commented:
- [ ] Note to self: once this is merged, create a pull request for https://github.com/zanata/zanata-server/compare/master...reduce-deploy-warnings

The changes can currently be seen in https://github.com/zanata/zanata-server/compare/wildfly10...reduce-deploy-warnings

seanf commented:
@carlosmunoz I'm thinking perhaps we should merge https://github.com/zanata/zanata-server/compare/wildfly10...reduce-deploy-warnings into this branch, allowing the changes to be tested together, instead of waiting to create another PR. What do you think?

carlosmunoz commented:
@seanf not sure of the details but I trust your judgement. Go for it.

seanf commented:
@carlosmunoz Could I get you to review the last three commits, please?

seanf commented:
@carlosmunoz Also, I don't think you reviewed https://github.com/zanata/zanata-server/commit/373c39c799423395d7d0f566980f2864581a1089 yet.

carlosmunoz commented:
Reviewed last commits. And thanks for changing the wildfly name btw.

seanf commented:
@zanata-jenkins test this please

seanf commented:
@zanata-jenkins test this please

djansen-redhat commented on docs/release-notes.md:
```diff
@@ -33,6 +33,9 @@
 * [ZNTA-793](https://zanata.atlassian.net/browse/ZNTA-793) - Remove unused method in TranslationMemoryAction
 
+<h5>Infrastructure Changes</h5>
+* Zanata now requires JBoss EAP 6.4.6.GA or later (recommended), or WildFly version 10.x.
```
Requires, but recommended? Or is "later than 6.4.6.GA" recommended?

seanf replied:
Well, the idea is that EAP is recommended (6.4.6 or later), but WildFly 10
is also... possible.

On 16 March 2016 at 13:51, Damian Jansen <notifications@github.com> wrote:

> In docs/release-notes.md
> <https://github.com/zanata/zanata-server/pull/990#discussion_r56278787>:
>
> > @@ -33,6 +33,9 @@
> >  * [ZNTA-928](https://zanata.atlassian.net/browse/ZNTA-928) - Readonly project doesn't have "lock" icon in UI
> >  * [ZNTA-793](https://zanata.atlassian.net/browse/ZNTA-793) - Remove unused method in TranslationMemoryAction
> >
> > +<h5>Infrastructure Changes</h5>
> > +* Zanata now requires JBoss EAP 6.4.6.GA or later (recommended), or WildFly version 10.x.
>
> Requires, but recommended? Or is "later than 6.4.6.GA" recommended?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/990/files/5ebc7bcb5d17ad1f2421243ecd6379f78e6142fd#r56278787>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

seanf replied:
@djansen-redhat See https://github.com/zanata/zanata-server/pull/990/commits/5442e0e25743cbfd489399cfe78cbd7e9a185d9f

djansen-redhat commented:
It says it requires wildfly 10, but the appserver profile is wildfly8

seanf commented:
It really means wildfly8+, because it was used for WildFly 8 and 9 as well.
It only affects developers, so we haven't bothered renaming it to 'wildfly'
yet. (Because changing it will knock-on to build scripts, CI jobs, etc.)

On 16 March 2016 at 14:05, Damian Jansen <notifications@github.com> wrote:

> It says it requires wildfly 10, but the appserver profile is wildfly8
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/990#issuecomment-197142792>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

djansen-redhat commented:
@seanf I'm getting [this error](http://pastebin.test.redhat.com/357447) when starting the server under Wildfly 10.

seanf commented:
> @seanf I'm getting this error when starting the server under Wildfly 10.

@djansen-redhat Did you try again on eap 6? 

I might need you to email me the details of the database you were testing with, perhaps including the contents of the DBCHANGELOG table.

djansen-redhat commented:
@seanf  It appears to works fine on EAP6.4. Will email.

seanf commented:
Looks like we have a problem with the standalone.xml in zanata-overlay for wildfly 10. I'll have to investigate.

seanf commented:
@carlosmunoz comparing the old (wildfly 9) config with the wildfly 10 config in zanata-overlay, there seem to be quite a few changes.

Did you mean to add the system property `document.storage.directory`? Also, the document-storage-directory JNDI string has changed from ${user.home}/zanata/files to ${user.dir}/zanata/documents. From what I can tell, MigrateRawDocumentsToFileSystem uses the JNDI string if set, but tries to fall back on the system property otherwise.  The system property is probably only intended for command-line invocations of Liquibase.

It seems like any existing WildFly installations may lose access to their uploaded documents, unless they move the directory contents around. 

The logging format has changed (used to use UTC timestamps, now it will use local time).

The 'hibernate' cache-container seems to have lost its jndi-name. I thought we were referring to this in persistence.xml, so I'm not sure how it can work.

The JNDI string java:global/zanata/smtp/port was added, but I don't think it is used anywhere in Zanata these days.

Finally, the socket bindings have changed a lot. They seem to be exactly the same as used in functional tests, which means the ports are offset by 100, except smtp which is port 2552.

I'm thinking of reverting most if not all of the above changes, unless you object.

carlosmunoz commented:
> Did you mean to add the system property document.storage.directory? 

I don't think I had anything to do with that.

> Also, the document-storage-directory JNDI string has changed from ${user.home}/zanata/files to ${user.dir}/zanata/documents. From what I can tell, MigrateRawDocumentsToFileSystem uses the JNDI string if set, but tries to fall back on the system property otherwise. The system property is probably only intended for command-line invocations of Liquibase.

But isn't that just a settings change? All those using the previous value should not need to change it.

> The 'hibernate' cache-container seems to have lost its jndi-name. I thought we were referring to this in persistence.xml, so I'm not sure how it can work.

This was probably a remnant of when we were using Infinispan for hibernate caching, which we don't do. You can ignore all modifications of the infinispan section **for hibernate** if you wish.

> The JNDI string java:global/zanata/smtp/port was added, but I don't think it is used anywhere in Zanata these days.

I think that's obsolete now since we configure the smtp settings directly on the mail session in standalone.xml

> Finally, the socket bindings have changed a lot. They seem to be exactly the same as used in functional tests, which means the ports are offset by 100, except smtp which is port 2552.

That doesn't make sense. They should be set to the default that come from wildfly.

> I'm thinking of reverting most if not all of the above changes, unless you object.

I think I'm ok. I think the reason for this might have been basing the xml on a working functional test one, together with some unused and/or leftover settings.

seanf commented:
It looks like the change in liquibase's checksum for changeset `db/changelogs/db.changelog-3.1.xml::6::damason@redhat.com` was caused by the creation of the system property `document.storage.directory`. This system property is referenced in `db/changelogs/db.changelog-3.1.xml` and passed as a CustomChange parameter. The parameters to a CustomChange are included in the checksum (by default), so the checksum changed.

I have removed the new system property and reverted a few other changes to make the wildfly 10 config closer to the previous wildfly 9 config.

seanf commented:
On 18 March 2016 at 15:38, Carlos Munoz <notifications@github.com> wrote:

> Did you mean to add the system property document.storage.directory?
>
> I don't think I had anything to do with that.
>
> Also, the document-storage-directory JNDI string has changed from
> ${user.home}/zanata/files to ${user.dir}/zanata/documents. From what I can
> tell, MigrateRawDocumentsToFileSystem uses the JNDI string if set, but
> tries to fall back on the system property otherwise. The system property is
> probably only intended for command-line invocations of Liquibase.
>
> But isn't that just a settings change? All those using the previous value
> should not need to change it.
>

I'm assuming that end users will run the installer again, because with the
messaging subsystem changes, their old wildfly 9 config won't work. If they
generate new config with different settings, the behaviour of Zanata will
change.


> The 'hibernate' cache-container seems to have lost its jndi-name. I
> thought we were referring to this in persistence.xml, so I'm not sure how
> it can work.
>
> This was probably a remnant of when we were using Infinispan for hibernate
> caching, which we don't do. You can ignore all modifications of the
> infinispan section *for hibernate* if you wish.
>
> The JNDI string java:global/zanata/smtp/port was added, but I don't think
> it is used anywhere in Zanata these days.
>
> I think that's obsolete now since we configure the smtp settings directly
> on the mail session in standalone.xml
>
> Finally, the socket bindings have changed a lot. They seem to be exactly
> the same as used in functional tests, which means the ports are offset by
> 100, except smtp which is port 2552.
>
> That doesn't make sense. They should be set to the default that come from
> wildfly.
>
> I'm thinking of reverting most if not all of the above changes, unless you
> object.
>
> I think I'm ok. I think the reason for this might have been basing the xml
> on a working functional test one, together with some unused and/or leftover
> settings.
>

True.

> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/990#issuecomment-198216283>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

seanf commented:
@carlosmunoz can you please take a look at https://github.com/zanata/zanata-server/pull/990/commits/2daaba75f432b1c07b1a1aeb55324d624362414f ?

carlosmunoz commented:
2daaba7 reviewed :+1:

djansen-redhat commented:
:white_check_mark: tested

seanf commented:
Squash-merged as https://github.com/zanata/zanata-server/commit/5b2d8ca892cdd4e4caccb7845ba4a14c31bf4ee4
